src/Microsoft.OData.Core/Value/ODataBinaryStreamValue.cs

```diff
@@ -0,0 +1,32 @@
+//---------------------------------------------------------------------
+// <copyright file="ODataBinaryStreamValue.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData
+{
+    #region Namespaces
+    using System.IO;
+    #endregion
+
+    /// <summary>
+    /// A class to represent a binary stream value
+    /// </summary>
+    public sealed class ODataBinaryStreamValue : ODataValue
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="stream">Input stream</param>
+        public ODataBinaryStreamValue(Stream stream)
+        {
+            this.Stream = stream;

```

Comment:
>this.Stream = stream; [](start = 12, length = 21)

shall we check the input stream nullable?

Reply:
Added an check here, and asserts in the other (internal) wrapping classes.

---
In reply to: [268350996](https://github.com/OData/odata.net/pull/1204#discussion_r268350996) [](ancestors = 268350996)